bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/SolarmanLoggerHandler.java

```diff
@@ -0,0 +1,222 @@
+import org.openhab.binding.solarman.internal.state.LoggerState;
+import org.openhab.binding.solarman.internal.updater.SolarmanChannelUpdater;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolarmanLoggerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Catalin Sanda - Initial contribution
+ */
+@NonNullByDefault
+public class SolarmanLoggerHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(SolarmanLoggerHandler.class);
+
+    private final DefinitionParser definitionParser;
+    private final SolarmanChannelManager solarmanChannelManager;
+    private final LoggerState loggerState;
+    @Nullable
+    private volatile ScheduledFuture<?> scheduledFuture;
+
+    public SolarmanLoggerHandler(Thing thing) {
+        super(thing);
+        this.definitionParser = new DefinitionParser();
+        this.solarmanChannelManager = new SolarmanChannelManager();
+        this.loggerState = new LoggerState();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        SolarmanLoggerConfiguration config = getConfigAs(SolarmanLoggerConfiguration.class);
+        SolarmanLoggerConnector solarmanLoggerConnector = new SolarmanLoggerConnector(config);
+
+        List<Channel> staticChannels = thing.getChannels().stream()
+                .filter(channel -> !channel.getProperties().containsKey(DYNAMIC_CHANNEL)).toList();
+
+        InverterDefinition inverterDefinition = definitionParser.parseDefinition(config.inverterType);
+
+        if (inverterDefinition == null) {
+            logger.error("Unable to find a definition for the provided inverter type");
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Unable to find a definition for the provided inverter type");
+            return;
+        } else {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found definition for {}", config.inverterType);
+            }
+        }
+        SolarmanV5Protocol solarmanV5Protocol = new SolarmanV5Protocol(config);
+
+        @Nullable
+        String additionalRequests = config.getAdditionalRequests();
+        List<Request> mergedRequests = StringUtils.isNotEmpty(additionalRequests)
+                ? mergeRequests(inverterDefinition.getRequests(),
+                        extractAdditionalRequests((@NonNull String) additionalRequests))
```

Comment:
`@NonNull` should not be needed as the class is already annotated with `@NonNullByDefault`